### PR TITLE
Add user auth support azure waf

### PR DESCRIPTION
**Pull request by YuvHayun**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-3695](https://jira-hq.paloaltonetworks.local/browse/CIAC-3695)

## Description
Added support for user-auth authentication flow.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

### CODE REVIEW

File: Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py [hunk 1]
```diff
@@ -332,6 +345,23 @@ def reset_auth(client: AzureWAFClient):
                                           '**!azure-waf-auth-start** and **!azure-waf-auth-complete**.')
 
 
+@logger
+def test_module(client):
+    """
+    Performs basic GET request to check if the API is reachable and authentication is successful.
+    Returns ok if successful.
+    """
+    if demisto.params().get('auth_type') == 'Device':
```
Comment (JudahSchwartz) [inline Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py:354]: we cant run azure-waf-auth-test here?

File: Packs/AzureWAF/Integrations/AzureWAF/AzureWAF_description.md [hunk 2]
```diff
@@ -1,14 +1,36 @@
...
+## Required Permissions:
+1. user_impersonation
+2. offline_access
+3. user.read 
+
+## Authentication Using the  User-Authentication Flow (recommended)
+
+Follow these steps for a self-deployed configuration:
+
+1. To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal. To add the registration, refer to the following [Microsoft article](https://docs.microsoft.com/en-us/microsoft-365/security/defender/api-create-app-web?view=o365-worldwide#create-an-app) steps 1-8.
```
Comment (JudahSchwartz) [inline Packs/AzureWAF/Integrations/AzureWAF/AzureWAF_description.md:13]: isnt this a duplicate of a few lines above?

File: Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py [hunk 3]
```diff
@@ -112,12 +125,12 @@ def delete_policy(self, policy_name: str, resource_group_name: str):
...
-    client.ms_client.get_access_token()  # If fails, MicrosoftApiModule returns an error
-    # should return error if user does not have permissions for the service.
-    client.get_policy_list_by_resource_group_name(client.resource_group_name)
-    return '✅ Great Success!'
+    if demisto.params().get('auth_type') == 'Device':
+        client.get_policy_list_by_resource_group_name(client.resource_group_name)  # If fails, MicrosoftApiModule returns an error
+        return CommandResults(readable_output='✅ Success!')
+    else:
+        client.ms_client.get_access_token()  # If fails, MicrosoftApiModule returns an error
+        return CommandResults(readable_output='✅ Success!')
```
Comment (JudahSchwartz) [inline Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py:133]:
```suggestion
    if demisto.params().get('auth_type') == 'Device':
        client.get_policy_list_by_resource_group_name(client.resource_group_name)  # If fails, MicrosoftApiModule returns an error
    else:
        client.ms_client.get_access_token()  # If fails, MicrosoftApiModule returns an error
return CommandResults(readable_output='✅ Success!')
```

Comment (JudahSchwartz) [reply Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py:133]: not sure if this came out normal or not. I just moved `return CommandResults(readable_output='✅ Success!')` out of the if else block

File: Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py [hunk 4]
```diff
@@ -27,9 +27,18 @@
 
 class AzureWAFClient:
     @logger
-    def __init__(self, app_id, subscription_id, resource_group_name, verify, proxy,
+    def __init__(self, app_id, subscription_id, resource_group_name, verify, proxy, auth_type, tenant_id=None,
+                 enc_key=None, auth_code=None, redirect_uri=None,
                  azure_ad_endpoint: str = 'https://login.microsoftonline.com'):
-
+        AUTH_TYPES_DICT: dict = {'User Auth': {
```
Comment (JudahSchwartz) [inline Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py:33]: can we format a bit differently so its clearer what the keys are? like
```
{
    key1 : { ...
           ...
           ...},
    key2: { ...
           ...
           ...}
 }
```
 if the linters let

**Review by JudahSchwartz — COMMENTED** [review]
looks great! some minor things

File: Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py (hunk 1 above)
Comment (YuvHayun) [reply Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py:354]: No, You need to configure the instance first.. we already tried that

File: Packs/AzureWAF/Integrations/AzureWAF/AzureWAF_description.md (hunk 2 above)
Comment (YuvHayun) [reply Packs/AzureWAF/Integrations/AzureWAF/AzureWAF_description.md:13]: I don't think so

**Review by JudahSchwartz — APPROVED** [review]

File: Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.yml [hunk 5]
```diff
@@ -19,6 +19,37 @@ configuration:
   name: resource_group_name
   required: true
   type: 0
+- additionalinfo: Type of authentication - could be Authorization Code Flow (recommended) or Device Code Flow.
+  display: Authentication Type
+  name: auth_type
+  required: false
```
Comment (yucohen) [inline Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.yml:25]: Lets make it true (in all other PRs too) with a default vale
@JudahSchwartz means we will force merge

**Review by yucohen — APPROVED** [review]

**Comment by YuvHayun** [comment]
@ShahafBenYakir I'll need force merge on this PR due to `required:true` on the **auth-type** param (which is part of the design).

**Pull request merged**